### PR TITLE
Adds Correct Handling for Unicode when calling API

### PR DESCRIPTION
**Pull request by JeffBain**
* We now properly encode our data when speaking with the SDE API

Refs SDE-1795

### CODE REVIEW

**Comment by JeffBain** [comment]
This should now have passing tests and some tests for our Unicode methods.

File: sdetools/sdelib/restclient.py [hunk 1]
```diff
@@ -161,8 +161,39 @@ def post_conf_init(self):
 
         self._post_init_done = True
 
+    def encode_string_utf8(self, string):
+        return string.encode('utf-8')
+
+    def encode_dict_utf8(self, args):
```
Comment (hanifvirani) [inline sdetools/sdelib/restclient.py]:
```python
return dict((encode_string_utf8(k), encode_string_utf8(v)) for k, v in args.items())
```

File: sdetools/sdelib/restclient.py [hunk 2]
```diff
@@ -161,8 +161,39 @@ def post_conf_init(self):
...
+        for key, value in args.iteritems():
+            if isinstance(key, unicode):
+                key = key.encode('utf-8')
+            if isinstance(value, unicode):
+                value = value.encode('utf-8')
+            encoded_args[key] = value
+        return encoded_args
+
+    def encode_list_utf8(self, args_list):
+        encoded_list = []
```
Comment (hanifvirani) [inline sdetools/sdelib/restclient.py]: [i.encode_string_utf8 for i in args_list]

Comment (hanifvirani) [inline sdetools/sdelib/restclient.py]: Small comments. Not fully aware of the issue but the code looks fine otherwise.

Comment (JeffBain) [inline sdetools/sdelib/restclient.py]: Good catches. I changed how the string thing worked and didn't catch that this would work now.

**Comment by emintham** [comment]
nice clean tests. :+1:

**Comment by geoffwhittington** [comment]
+1

**Pull request merged**